src/Toolchain/ToolchainProvider.ts

```diff
@@ -161,7 +166,25 @@ export class ToolchainProvider implements vscode.TreeDataProvider<ToolchainNode>
     });
   }
 
-  run(cfg: string) {
-    throw Error('Not implemented yet');
+  run(cfg: string, toolchain?: Toolchain) {
+    // TODO(jyoung) Implement default backend
+    let activeToolchain = toolchain ? toolchain : gToolchainEnvMap['tv2'].default();
+    if (!activeToolchain) {
+      Logger.error(this.tag, 'Toolchain is not installed');
+      vscode.window.showErrorMessage('Toolchain is not installed');
+      return;
+    }
+
+    Logger.info(this.tag, `Compile toolchain: ${activeToolchain.info.name}-${activeToolchain.info.version?.str()}`);
+    // TODO(jyoung) Implement default backend
+    gToolchainEnvMap['tv2']
+        .run(cfg, activeToolchain)
+        .then(
+            () => {
+              vscode.window.showInformationMessage('Compile successfully');
```

Comment:
How about (since 'Compile successfully` sounds like imperative sentence)?

```suggestion
              vscode.window.showInformationMessage('Successfully Compiled.');
```